src/watchdog/GOPATH/src/github.com/microsoft/watchdog/cmd/watchdog/main.go

```diff
@@ -0,0 +1,78 @@
+// MIT License
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE
+
+package main
+
+import (
+	"flag"
+	"net/http"
+	"strconv"
+	"time"
+
+	"k8s.io/klog"
+
+	"github.com/microsoft/watchdog/pkg/watchdog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	collectionInterval int
+	gcInterval         int
+	port               int
+)
+
+func init() {
+	klog.InitFlags(nil)
+	flag.Set("v", "2")
+
+	flag.IntVar(&collectionInterval, "collection-interval", 180, "Interval between two collections (seconds)")
+	flag.IntVar(&gcInterval, "gc-interval", 60, "Interval between two garbage collections (minutes)")
```

Comment:
Do we need to collect (call APIServer) every 3mins? Is it too frequent?

Reply:
By default it will collect every 3mins, This changed by this PR: https://github.com/microsoft/pai/pull/3761, before that, we collect metrics every 30 seconds...